Yousei.Connectors/Telegram/TelegramConnection.cs

```diff
@@ -35,10 +35,13 @@ private IObservable<Unit> CreateConnectionObservable(TelegramBotClient telegramB
                 .RefCount(TimeSpan.FromSeconds(1));
 
         private IObservable<T> WrapEvent<T>(IObservable<Unit> connectionObservable, object target, string eventName)
-                    where T : class
-        {
-            var eventObservable = Observable.FromEventPattern<T>(target, eventName).Select(o => o.EventArgs);
-            return Observable.Merge(connectionObservable.Select(_ => default(T)), eventObservable);
-        }
+            where T : class
+            => Observable.DeferAsync(cancellationToken =>
+                {
+                    var subscription = connectionObservable.Subscribe(_ => { });
```

Comment:
Why does this subscribe to basically an empty function?

Reply:
I did not find another way to link the lifetime to another subscription. This observable will start the receive loop when the first observer subscribes to it and stops when the last one unsubscribes.